components/motor/gpio/basic.go

```diff
@@ -190,35 +197,33 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 
 	switch m.motorType {
 	case ABPwm, DirectionPwm:
-		if math.Abs(powerPct) <= 0.001 {
+		if pwmPower <= 0.001 {
 			return m.turnOff(ctx, extra)
 		}
 		pwmPin = m.PWM
 	case AB:
 		switch {
-		case powerPct >= 0.001:
+		case pwmPower >= 0.001:
 			pwmPin = m.B
 			if m.dirFlip {
 				pwmPin = m.A
 			}
-			powerPct = 1.0 - math.Abs(powerPct) // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
-		case powerPct <= -0.001:
+			pwmPower = 1.0 - pwmPower // Other pin is always high, so only when PWM is LOW are we driving. Thus, we invert here.
+		case pwmPower <= -0.001:
```

Comment:
isn't there an issue here now that pwmPower is now an absolute value?

Reply:
nice catch! I'll ad a minus sign